cli/gardener_ci/productutil.py

```diff
@@ -195,19 +278,23 @@ def add_dependencies(
             c=component_name,
             v=component_version,
             f=descriptor_src_file
-        )
-        )
-
-    component_deps = component.dependencies()
-
-    for component_ref in component_dependencies:
-        component_deps.add_component_dependency(component_ref)
-    for image_dep in container_image_dependencies:
-        component_deps.add_container_image_dependency(image_dep)
-    for web_dep in web_dependencies:
-        component_deps.add_web_dependency(web_dep)
-    for generic_dep in generic_dependencies:
-        component_deps.add_generic_dependency(generic_dep)
+        ))
+
+    # maintain old behaviour
+    if not validation_policies:
+        validation_policies=[ValidationPolicy.FORBID_EXTRA_ATTRIBUTES]
+
+    _add_dependencies(
```

Comment:
as discussed:

split this into two functions:

- parse_dependencies (which returns a sequence (e.g. a generator, but this is up to you, of course)) of dependency-model-objects
- add_dependencies (which accepts said sequence + validation options + target component descriptor)

add-dependencies should handle the proper dispatching of the dependency-objects, depending on type. as discussed, it might make sense to add such a method to the dependencies model class